src/cfnlint/rules/resources/properties/ValuePrimitiveType.py

```diff
@@ -46,16 +61,35 @@ def initialize(self, cfn):
         for property_spec in self.property_specs:
             self.resource_sub_property_types.append(property_spec)
 
+    def _value_non_strict(self, value, path, item_type):
+        """ Checks non strict """
+        matches = []
+        try:
+            if item_type in ['String']:
+                str(value)
+            elif item_type in ['Boolean']:
+                bool(value)
+            elif item_type in ['Integer']:
+                int(value)
+        except Exception:  # pylint: disable=W0703
+            message = 'Property %s should be of type %s' % ('/'.join(map(str, path)), item_type)
+            matches.append(RuleMatch(path, message))
+
+        return matches
+
     def check_primitive_type(self, value, item_type, path):
         """Chec item type"""
         matches = []
 
         if isinstance(value, dict) and item_type == 'Json':
             return matches
         if item_type in ['String']:
-            if not isinstance(value, (str, six.text_type, six.string_types)):
-                message = 'Property %s should be of type String' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+            if not isinstance(value, (six.string_types)):
+                if self.config['strict'] is False:
```

Comment:
Isn't `if not self.config['strict']` more "pythonic"? (Not sure here, just a question)